src/main/java/io/temporal/samples/encryptedpayloads/CryptDataConverter.java

```diff
@@ -106,47 +110,39 @@ private SecretKey getKey() {
 
     Payload innerPayload = optionalPayload.get();
 
-    ByteString encoding = innerPayload.getMetadataOrDefault(EncodingKeys.METADATA_ENCODING_KEY, null);
-    if (encoding == null) {
-      return optionalPayload;
-    }
-
     byte[] encryptedData;
     try {
-      encryptedData = encrypt(innerPayload.getData().toByteArray());
+      encryptedData = encrypt(innerPayload.toByteArray());
     } catch (Throwable e) {
       throw new DataConverterException(e);
     }
 
-    Payload encryptedPayload = Payload.newBuilder(innerPayload)
-      .setData(ByteString.copyFrom(encryptedData))
-      .putMetadata(EncodingKeys.METADATA_ENCODING_KEY, EncodingKeys.METADATA_ENCODING_RAW)
-      .putMetadata(EncodingKeys.METADATA_WRAPPED_ENCODING_KEY, encoding)
-      .build();
+    Payload encryptedPayload =
+        Payload.newBuilder()
+            .putMetadata(METADATA_ENCODING_KEY, METADATA_ENCODING)
```

Comment:
I would add cipher and some sort of key id into the metadata. The idea is that the metadata should be enough to find out which key to use.